### PR TITLE
Fixing issue #1315

### PR DESCRIPTION
**Pull request by Ramaraju-1996**
The file changes try to fix issue #1315(Document openid implementation better).

### CODE REVIEW

**Comment by BelieveC** [comment]
Nicely done, @Ramaraju-1996 Check [this](https://hakiri.io/github/openSUSE/osem/master/ffb0e256890dc2f5df8bade156982cb4954ff6b1) Hakiri security error.

**Comment by Ramaraju-1996** [comment]
Thnx @BelieveC , I have checked that error details but I am unable to understand what the actual error is about.

**Comment by Ana06** [comment]
The error is unrelated to this PR, as this is just text and it is also failing in other PRs. :wink:

**Comment by Ramaraju-1996** [comment]
Tnx @Ana06,
But is there any work-around or solution to this problem?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -107,3 +107,20 @@ config.ichain_force_test_username = "testuser"
...
 ```
+
+### Using openID in development mode  
+In order to use [openID](http://openid.net/) logins for your OSEM installation you need to register your application with the providers ([Google](https://code.google.com/apis/console#:access), [GitHub](https://github.com/settings/applications/new) or [Facebook](https://developers.facebook.com/)) and enter their API keys in `config/secrets.yml` file, changing the existing sample values.  
+####Note:  
+1.Since we are taking into consideration the 'development' environment, please do the replacement in the below steps under the 'development' section.  
+2.For any other environment, please do the replacement in the below steps under the respective environment sections.  
+####Steps to be followed:  
+1.Create 'config/secrets.yml' file.  
+2.'config/secrets.yml' file has to be similar to the 'config/secrets.yml.example' file.
```
Comment (ChrisBr) [inline CONTRIBUTING.md]: Merge step 1. & 2. to ``cp config/secrets.yml.example config/secrets.yml``

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -107,3 +107,20 @@ config.ichain_force_test_username = "testuser"
...
+
+### Using openID in development mode  
+In order to use [openID](http://openid.net/) logins for your OSEM installation you need to register your application with the providers ([Google](https://code.google.com/apis/console#:access), [GitHub](https://github.com/settings/applications/new) or [Facebook](https://developers.facebook.com/)) and enter their API keys in `config/secrets.yml` file, changing the existing sample values.  
+####Note:  
+1.Since we are taking into consideration the 'development' environment, please do the replacement in the below steps under the 'development' section.  
+2.For any other environment, please do the replacement in the below steps under the respective environment sections.  
+####Steps to be followed:  
+1.Create 'config/secrets.yml' file.  
+2.'config/secrets.yml' file has to be similar to the 'config/secrets.yml.example' file.  
+3.Run 'rake secret' and note down the secret key base.
```
Comment (ChrisBr) [inline CONTRIBUTING.md]: Rake secret is not really necessary for development. However, it doesn't hurt

**Comment by ChrisBr** [comment]
@differentreality please also have a look, not really familiar with openID and if all steps are really necessary...

**Comment by ChrisBr** [comment]
@Ramaraju-1996 the hakiri issue is fixed in https://github.com/openSUSE/osem/pull/1323, so please rebase and it should also resolve the failure in this PR

**Comment by differentreality** [comment]
Actually, we have those variables in dotenv now, so all relevant info needs to be updated. This PR is not really doing that all.

**Comment by differentreality** [comment]
We are currently using dotenv for configuration, and it is not documented in your changes. Moreover this is not explaining what the user can do with openid in development / how things will work. I will close this, as it needs quite some re-thinking and re-writing, which is why #1315 was created.

**Pull request closed without merging**